File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPInterstitialAdView.java
package com.monet.bidder;

import android.content.Context;

import com.google.android.gms.ads.InterstitialAd;

class DFPInterstitialAdView implements AdServerAdView {
  private final InterstitialAd interstitialAd;
  private String mAdUnitId;

  DFPInterstitialAdView(InterstitialAd interstitialAd) {
    this.interstitialAd = interstitialAd;
    mAdUnitId = interstitialAd.getAdUnitId();
  }

  @Override
  public AdServerWrapper.Type getType() {
    return AdServerWrapper.Type.INTERSTITIAL;
  }

  @Override
  public String getAdUnitId() {
    return mAdUnitId;
  }

  @Override
  public void setAdUnitId(String adUnitId) {
    mAdUnitId = adUnitId;
  }

  @Override
  public Context getContext() {
    return null;
  }

  @Override
  public void loadAd(AdServerAdRequest request) {
    DFPAdViewRequest adRequest = (DFPAdViewRequest) request;
    if(interstitialAd != null){
      interstitialAd.loadAd(adRequest.getDFPRequest());
    }
  }
}

File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPInterstitialAdView.kt
package com.monet.bidder

import android.annotation.SuppressLint
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.InterstitialAd
import com.monet.AdServerAdRequest
import com.monet.AdServerAdView
import com.monet.AdType
import com.monet.AdType.INTERSTITIAL
import com.monet.DFPAdViewRequest

internal class DFPInterstitialAdView(private val interstitialAd: InterstitialAd?) : AdServerAdView {
  override var adUnitId: String = interstitialAd?.adUnitId ?: ""
  override var type: AdType = INTERSTITIAL

  @SuppressLint("MissingPermission")
  override fun loadAd(request: AdServerAdRequest?) {
    request?.let {
      val adRequest = it as DFPAdViewRequest
      interstitialAd?.loadAd(adRequest.adRequest.request as AdRequest)
    }
  }
}